tests/test_culling.py

```diff
@@ -333,14 +339,46 @@ def test_non_culled_host_is_not_removed(
     assert created_host_ids == [host.id for host in retrieved_hosts]
 
     threadctx.request_id = UNKNOWN_REQUEST_ID_VALUE
-    host_reaper_run(inventory_config, mock.Mock(), db.session, event_producer_mock)
+    host_reaper_run(
+        inventory_config,
+        mock.Mock(),
+        db.session,
+        event_producer_mock,
+        shutdown_handler=mock.Mock(**{"shut_down.side_effect": (False)}),
+    )
 
     retrieved_hosts = db_get_hosts(created_host_ids)
 
     assert created_host_ids == [host.id for host in retrieved_hosts]
     assert event_producer_mock.event is None
 
 
+@pytest.mark.host_reaper
+def test_reaper_shutdown_handler(
+    event_producer_mock, event_datetime_mock, db_create_host, db_get_hosts, inventory_config
+):
+    staleness_timestamps = get_staleness_timestamps()
+    created_hosts = []
+
+    for _ in range(3):
+        host = minimal_db_host(stale_timestamp=staleness_timestamps["culled"].isoformat(), reporter="some reporter")
+        created_hosts.append(db_create_host(host))
+
+    created_host_ids = [host.id for host in created_hosts]
+    assert db_get_hosts(created_host_ids)
```

Comment:
For symmetry and to be sure that we are doing the right thing, let’s assert the exact count here as well.

```suggestion
    retrieved_hosts = db_get_hosts(created_host_ids)
    assert retrieved_hosts.count() == 3
```